manager/src/main/java/org/openremote/manager/dashboard/DashboardResourceImpl.java

```diff
@@ -30,31 +31,68 @@ public DashboardResourceImpl(TimerService timerService, ManagerIdentityService i
 
     @Override
     public Dashboard[] getAllRealmDashboards(RequestParams requestParams, String realm) {
-
+        boolean publicOnly;
+        if(realm == null) {
+            throw new WebApplicationException(BAD_REQUEST);
+        }
         try {
+            // if not authenticated, or having no INSIGHTS access, only fetch public dashboards
+            if(isAuthenticated()) {
+                publicOnly = (!hasResourceRole(ClientRole.READ_INSIGHTS.getValue(), Constants.KEYCLOAK_CLIENT_ID));
+            } else {
+                publicOnly = true;
+            }
             if(!isRealmActiveAndAccessible(realm)) {
-                throw new WebApplicationException(FORBIDDEN);
+                publicOnly = true;
```

Comment:
If a realm is inactive then nothing should be retrievable from that realm

Reply:
Good spot!

Just noticed as well that `isRealmActiveAndAccessible()` will always return false if you're unauthenticated.
Should I do a manual `realm.getEnabled()` check instead?

Reply:
```
 public boolean isRealmActiveAndAccessible(AuthContext authContext, Realm realm) {
      if (realm == null) {
          return false;
      }

      boolean isSuperUser = authContext != null && authContext.isSuperUser();
      boolean isUsersRealm = isSuperUser || authContext == null || authContext.isRealmAccessibleByUser(realm.getName());

      return isSuperUser || (isUsersRealm && realm.isActive(timerService.getCurrentTimeMillis()));
  }
```

This shouldn't always return false if unauthenticated?